### PR TITLE
reprepro 5.3.0 (new formula)

### PR DESCRIPTION
**Pull request by cbarcenas**
- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

[reprepro](https://wiki.debian.org/DebianRepository/SetupWithReprepro) is a management tool for Debian package repositories.

Two minor things to note when reviewing this formula:

* Homebrew's keg-only `libarchive` is required, because macOS's built-in `libarchive` does not include the requisite C header files needed to build this package.

* This package must be compiled by GCC, because [Clang does not support GNU C nested functions](http://clang.llvm.org/docs/UsersManual.html#gcc-extensions-not-implemented-yet) which are used in a few places in this package.

### CODE REVIEW

**Comment by cbarcenas** [comment]
Bump :)

**Comment by leonklingele** [comment]
What's blocking a merge here?

@cbarcenas reprepro 5.3.0 has been released since:

```patch
diff --git a/Formula/reprepro.rb b/Formula/reprepro.rb
index cfc03f84bc..5add6c2594 100644
--- a/Formula/reprepro.rb
+++ b/Formula/reprepro.rb
@@ -2,8 +2,8 @@ class Reprepro < Formula
   desc "Debian package repository manager"
   homepage "https://salsa.debian.org/brlink/reprepro"

-  url "https://deb.debian.org/debian/pool/main/r/reprepro/reprepro_5.2.0.orig.tar.gz"
-  sha256 "bd3476f2c5523e5bea112657f850224c9b18ae04d2141ce3c3b589a9038d5c75"
+  url "https://deb.debian.org/debian/pool/main/r/reprepro/reprepro_5.3.0.orig.tar.gz"
+  sha256 "5a5404114b43a2d4ca1f8960228b1db32c41fb55de1996f62bc1b36001f3fab4"

   depends_on "berkeley-db@4"
   depends_on "gpgme"
```

**Comment by cbarcenas** [comment]
Hi @leonklingele, I don't think anything is blocking the merge.. I think the Homebrew maintainers simply haven't reviewed this PR yet.

Thank you for notifying me that upstream has been updated recently. I have updated the PR accordingly!

File: Formula/reprepro.rb
Comment (javian) [inline Formula/reprepro.rb]: please remove this empty line

**Comment by javian** [comment]
I think that this PR has just drowned in everything else that's been in here but we are close and should be able to merge it soon.

File: Formula/reprepro.rb
Comment (fxcoudert) [inline Formula/reprepro.rb]: Does this not support the latest `berkeley-db`?

File: Formula/reprepro.rb
Comment (fxcoudert) [inline Formula/reprepro.rb]: Yuck, installing GCC as hard dependency for such a simply software seems very heavy :(

**Review by fxcoudert — DISMISSED** [review]

File: Formula/reprepro.rb
Comment (cbarcenas) [reply Formula/reprepro.rb]: Yeah, it's unfortunate that upstream uses certain GNU C extensions that aren't implemented in other compilers. :(

Thankfully it's just a build-time dependency, so it won't affect most people installing the formula. The heavy build-time dependencies are one of my main motivations for packaging this for Homebrew.

File: Formula/reprepro.rb
Comment (cbarcenas) [reply Formula/reprepro.rb]: Nope, there was a backwards-incompatible change of the on-disk format used by BDB 4 vs 5+:

https://stackoverflow.com/questions/49354902/berkeley-db-4-and-later-versions-compatibility

There's still a few major OSS projects that are stuck with BDB 4; the most notable one in Homebrew is probably [`bitcoin`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/bitcoin.rb#L23).

**Comment by javian** [comment]
@BrewTestBot test this please

File: Formula/reprepro.rb
Comment (fxcoudert) [reply Formula/reprepro.rb]: GCC is not a build dependency, it's a runtime dep:

```
$ otool -L /usr/local/Cellar/reprepro/5.3.0/bin/reprepro | grep gcc
	/usr/local/lib/gcc/8/libgcc_s.1.dylib (compatibility version 1.0.0, current version 1.0.0)
```

since code compiled with GCC has to be linked against its runtime library.

Please add `depends_on "gcc"` to make it clearer.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

File: Formula/reprepro.rb
Comment (cbarcenas) [reply Formula/reprepro.rb]: Wow, TIL about `libgcc`. Thanks for catching this @fxcoudert, I've added `depends_on "gcc"` here.

I reviewed existing homebrew-core formulas that require GCC for compilation, and it seems like simply adding a runtime `gcc` dependency is indeed the current convention.

I read about libgcc in the GCC manual (https://gcc.gnu.org/onlinedocs/gccint/Libgcc.html), and learned that it is very lightweight, and primarily provides implementations of mathematical operations that are not supported on a target architecture's hardware plus some other C++ exception-handling functionality.

I think an alternative to adding a hard runtime dependency on `gcc` in many Homebrew packages such as this one would be to add `-static-libgcc` to `LDFLAGS`.

Static linking libgcc could be appropriate because the library is extremely lightweight, and in some cases may even be strictly better in terms of simplifying generated binaries: for example, for this formula, I tested compiling with and without `-static-libgcc` and there didn't seem to be any effect on compiled code size. In fact, even though `/usr/local/lib/gcc/8/libgcc_s.1.dylib` is listed as a dylib dependency, `nm -m` didn't list any symbols to be imported from this dylib at all at runtime!

Thoughts / should I ticket this as a possible change in Homebrew itself?

Comment (fxcoudert) [reply Formula/reprepro.rb]: Static libgcc increases the size of all binaries and libraries, so it needs to be a case-by-case decision, not something that Homebrew does automatically behind your back.

But if the software can detect and build with a static libgcc (say, with a configure flag) then yes, in that case we could have `gcc` as a build dependency, which would be ideal for users.

**Review by fxcoudert — APPROVED** [review]

**Comment by fxcoudert** [comment]
Thanks @cbarcenas for your contribution to Homebrew!

**Pull request closed without merging**
